server/tabby_server/vision/image_labelling.py

```diff
@@ -0,0 +1,187 @@
+from contextlib import contextmanager
+from ultralytics import YOLO
+import torch
+import json
+import os
+
+"""
+Enables You Only Look Once (YOLO) image recognition.
+
+This will take in an image and scan it for books before outputting
+a json with locations, bounding boxes, and confidence.
+
+This is unroutable and onlt meant to be called by an endpoint.
+"""
+
+
+@contextmanager
+def temp_dir(path):
+    """
+    Context Manages to allow temporarily changing the working directory to
+    match the given path.
+
+    Ensures that the original path is restored even if the change fails or if
+    an exception occurs when this is in use.
+
+    Primarily to allow directly calling to this function for testing purposes.
+    """
+
+    original_dir = os.getcwd()
+    try:
+        os.chdir(path)
+        yield
+    finally:
+        os.chdir(original_dir)
+
+
+# Expected Image Size for Model to Use
+EXPECTED_SIZE = (640, 640)
+
+# Temporarily Change Working Directory
+with temp_dir("tabby_server/vision/"):
+    # Load pretrained model from other directory.
+    model = YOLO("shelf_yolo.pt")
+
+
+def find_books(tensor_image=None) -> dict[str, list]:
```

Comment:
The return type annotation is inconsistent with the documentation. `list of dicts` versus `dict[str, list]`

Reply:
Fixed.